modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQXAConnectionFactory.java

```diff
@@ -0,0 +1,168 @@
+ */
+package org.wso2.andes.client;
+
+import org.wso2.andes.url.URLSyntaxException;
+
+import javax.jms.JMSException;
+import javax.jms.JMSSecurityException;
+import javax.jms.XAConnection;
+import javax.jms.XAConnectionFactory;
+import javax.jms.XAQueueConnection;
+import javax.jms.XAQueueConnectionFactory;
+import javax.jms.XATopicConnection;
+import javax.jms.XATopicConnectionFactory;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * This class contains XA Connection Factory
+ * To initialize a XA Connection Factory,
+ * In the jndi file, we should define connection Factory with xaconnectionfactory prefix
+ * Example : xaconnectionfactory.QueueConnectionFactory = amqp://admin:admin@clientID/carbon?brokerlist='tcp://localhost:5675'
+ */
+public class AMQXAConnectionFactory extends AMQConnectionFactory implements XATopicConnectionFactory,
+        XAQueueConnectionFactory, XAConnectionFactory {
+
+    /**
+     * Scheduled executor share by the XAConnectionImpl objects
+     */
+    private ScheduledExecutorService scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+
+
+    public AMQXAConnectionFactory(String url) throws URLSyntaxException {
+        super(url);
+    }
+
+    /**
+     * Creates a XAConnection with the default user identity.
+     * <p> The XAConnection is created in stopped mode. No messages
+     * will be delivered until the <code>Connection.start</code> method
+
+     * is explicitly called.
+     *
+     * @return A newly created XAConnection
+     * @throws JMSException         If creating the XAConnection fails due to some internal error.
+     * @throws JMSSecurityException If client authentication fails due to an invalid user name or password.
+     */
+    public XAConnection createXAConnection() throws JMSException
+    {
+        try
+        {
+            return new XAConnectionImpl(_connectionDetails, _sslConfig, scheduledExecutor);
+        }
+        catch (Exception e)
+        {
+            JMSException jmse = new JMSException("Error creating connection: " + e.getMessage());
+            jmse.setLinkedException(e);
+            jmse.initCause(e);
+            throw jmse;
+        }
+    }
+
+    /**
+     * Creates a XAConnection with the specified user identity.
+     * <p> The XAConnection is created in stopped mode. No messages
+     * will be delivered until the <code>Connection.start</code> method
+     * is explicitly called.
+     *
+     * @param username the caller's user name
+     * @param password the caller's password
+     * @return A newly created XAConnection.
+     * @throws JMSException         If creating the XAConnection fails due to some internal error.
+     * @throws JMSSecurityException If client authentication fails due to an invalid user name or password.
+     */
+    public XAConnection createXAConnection(String username, String password) throws JMSException
+    {
+        if (_connectionDetails != null)
```

Comment:
let's remove '_' from _connectionDetails variable name

Reply:
@milindaperera: this is a forked code, therefore IMO It's not good to modify the code.

Reply:
+1